java/src/main/java/gherkin/ast/GherkinDocument.java

```diff
@@ -0,0 +1,25 @@
+package gherkin.ast;
+
+import java.util.Collections;
+import java.util.List;
+
+public class GherkinDocument extends Node {
```

Comment:
Extending Node is a little bit of an anomaly since `GherkinDocument` does not have a location as all other `Node`:s (thus the `super(null)` to set the location to null). I did not experiment with not having `GherkinDocument` extending `Node` (and explicitly declare a `type` field in `GherkinDocument`), or if that would not work, move the `Location` field from `Node` to an intermediate class between `Node` and all its sub classes that have a location.